Pages/shop.js

```diff
@@ -0,0 +1,207 @@
+const Page = require("./page");
+
+var EC = protractor.ExpectedConditions,
+
+    data = require("../Data/data.js");
+
+class Shop extends Page.Page{
+    constructor(){
+        super();
+        this.title="Shop"
+    }
+
+    // GETTERS
+
+    get shopGrid(){
+        return ".shop-page-grid";
+    }
+    
+    elementsInGridView(itemNumber){
+        return browser.driver.findElement(by.css(`.shop-page-grid :nth-child(${itemNumber})`))
+    }
+
+    get bidButton(){
+        return browser.driver.findElement(by.className("shop-bid-button"))
+    }
+
+    get bidButtonLocator(){
+        return ".shop-bid-button";
+    }
+
+    get listButton(){
+        return browser.driver.findElement(by.css(".shop-header-view span:last-child"));
+    }
+
+    get gridBidButton(){
+        return browser.driver.findElement(by.css(".overlay div"))
+    }
+    
+    filterNumber(number){
```

Comment:
Reduce code here. Can you make something like:
   filterNumber(number) {
if(number) { 
return browser.driver.findElement(by.css(`#shop-filter-cat-single:nth-child(${number})`));
} 
return browser.driver.findElement(by.css("#shop-filter-cat-single:nth-child(2)"));
}
}

Reply:
I made different logic on here, this filter(number) is only receiveing ${number}, a method that is calling filter(number) does the logic of controlling what is passed to filter(number).